packages/react/src/contents/hooks/__tests__/useSeofiles.test.tsx

```diff
@@ -0,0 +1,96 @@
+import { cleanup, renderHook } from '@testing-library/react';
+import { fetchSEOFiles } from '@farfetch/blackout-redux';
+import { mockInitialState } from './__fixtures__/useSeoMetadata.fixtures.js';
+import { seoFilesQuery as query } from 'tests/__fixtures__/contents/seoFiles.fixtures.mjs';
+import { useSeoFiles } from '../index.js';
+import { withStore } from '../../../../tests/helpers/index.js';
+
+jest.mock('@farfetch/blackout-redux', () => {
+  const original = jest.requireActual('@farfetch/blackout-redux');
+
+  return {
+    ...original,
+    fetchSEOFiles: jest.fn(() => ({ type: 'foo-bar' })),
+  };
+});
+
+describe('useSeoFiles', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  afterEach(() => cleanup());
+
+  it('should return null if there are no results', () => {
+    const props = { query };
+
+    const {
+      result: { current },
+    } = renderHook(() => useSeoFiles(props), {
+      wrapper: withStore(mockInitialState),
+    });
+
+    const emptyResult = {
+      error: undefined,
+      isLoading: undefined,
+      isFetched: false,
+      data: null,
+      actions: {
+        fetch: expect.any(Function),
+      },
+    };
+
+    expect(fetchSEOFiles).toHaveBeenCalledTimes(1);
+    expect(current).toEqual(emptyResult);
+  });
+
+  describe('options', () => {
+    it('should call fetch data if `enableAutoFetch` option is true', () => {
+      const props = { query, enableAutoFetch: true };
+
+      renderHook(() => useSeoFiles(props), {
+        wrapper: withStore(mockInitialState),
+      });
+
+      expect(fetchSEOFiles).toHaveBeenCalledWith({
+        hostId: 1234,
+        name: 'siteSEOFiles',
+        page: 1,
+        pageSize: 60,
+      });
+    });
+
+    it('should not fetch data if `enableAutoFetch` option is false', () => {
+      const props = { query, enableAutoFetch: false };
+
+      renderHook(() => useSeoFiles(props), {
+        wrapper: withStore(mockInitialState),
+      });
+
+      expect(fetchSEOFiles).not.toHaveBeenCalledWith();
+    });
+
+    describe('actions', () => {
+      it('should call `fetch` action', () => {
+        const {
+          result: {
+            current: {
+              actions: { fetch },
+            },
+          },
+        } = renderHook(() => useSeoFiles({ query }), {
```

Comment:
Set `enableAutoFetch` to false to prevent the `useEffect` call so you can be sure the only call made was from the `fetch` action you made.

Reply:
Resolved! ✅